README.dev.md

```diff
@@ -6,11 +6,612 @@ You will need a machine with at least 2GB of RAM available (see
 https://github.com/zulip/zulip/issues/32 for a plan for how to
 dramatically reduce this requirement).
 
-Start by cloning this repository: `git clone https://github.com/zulip/zulip.git`
+Contents:
+
+* [First-time contributors start here](#first-time-contributors-start-here)
+* [Using Vagrant](#using-vagrant)
+* [Using provision.py without Vagrant](#using-provisionpy-without-vagrant)
+* [By hand](#by-hand)
+* [Using Docker (experimental)](#using-docker-experimental)
+
+## First-time contributors start here
+
+This section guides first-time contributors through installing the Zulip dev
+environment on Windows 10, OS X El Capitan, Ubuntu 14.04, and Ubuntu 16.04. 
+
+Those who have installed Zulip before or are experienced at using Vagrant may
+wish to skip ahead to [Using Vagrant](#using-vagrant), [Using
+Docker](#using-docker-experimental), or read about installing [By
+hand](#by-hand).
+
+The recommended method for installing the Zulip dev environment is to use
+Vagrant with VirtualBox on Windows and OS X, and Vagrant with LXC on
+Ubuntu. This method creates a virtual machine (for Windows and OS X) 
+or a Linux container (for Ubuntu) inside which the Zulip server and
+all related services will run. 
+
+Contents:
+* [Requirements](#requirements)
+* [Step 1: Install Prerequisites](#step-1-install-prerequisites)
+* [Step 2: Get Zulip code](#step-2-get-zulip-code)
+* [Step 3: Start the dev environment](#step-3-start-the-dev-environment)
+* [Step 4: Developing](#step-4-developing)
+* [Troubleshooting & Common Errors](#troubleshooting--common-errors)
+
+If you encounter errors installing the Zulip dev environment and they are not addressed in [Troubleshooting & Common Errors](#troubleshooting--common-errors), send a note to the [Zulip-devel Google group](https://groups.google.com/forum/#!forum/zulip-devel) or [file an issue](https://github.com/zulip/zulip/issues).
+
+### Requirements
+
+Installing the Zulip dev environment requires downloading dependencies of
+several hundred megabytes. You will need an active internet connection
+throughout the entire installation processes.
+
+- **Windows**: Windows 10 64-bit, hardware virtualization enabled (VT-X or AMD-V), administrator access, [Cygwin][cygwin-dl], [VirtualBox][vbox-dl], [Vagrant][vagrant-dl].
```

Comment:
I'm a little uncomfortable with advertising this as "Windows 10" being required, since these instructions probably work fine on other Windows versions.  What do you think about softening this to "Windows 64-bit" at the start, and then at the end something like "Windows 10 is most carefully tested but instructions should work for other versions."?